### ISSUE TITLE
Duplicate moment instances when using moment.js in the project.

### ISSUE THREAD

**Issue by TarekAlQaddy** [brockpetrie/vue-moment#57]
I have a project in which I use `moment.js` and `vue-moment`, I found out that `vue-moment` requires `moment` again in it's own package in `node_modules`. In that way we have a duplicate resources.

I understand that this is required in case one does not use `moment.js` in the first place. But in my case it's really important to me to reduce the build size.

Is there any way to make `vue-moment` use the same `moment` resource I use?

**Comment by BrockReece** [comment]
Yep, you can pass a moment instance through when you install the plugin in your main.js

```js
// main.js
import Vue from 'vue'
import moment from 'moment'
import VueMoment from 'vue-moment'

Vue.use(VueMoment, {
  moment,
})
```

I hope that helps?

**Comment by TarekAlQaddy** [comment]
@BrockReece ,
Thank you for your concern, Unfortionatily That does not help the build sizes are the same in dev and prod .. I can still see in webpack bundle analyzer that vue moment uses it's own moment instance again

![webpack bundle analyzer](https://image.ibb.co/eqonQG/Webpack_Bundle_Analyzer.png)

**Comment by AustinPaquette** [comment]
@TarekAlQaddy did you ever solve this?

**Comment by TarekAlQaddy** [comment]
@AustinPaquette No, I didn't find a way to do it.

**Comment by brockpetrie** [comment]
@AustinPaquette Does running `npm dedupe` fix this?

In old npm versions, we'd be able to use peerDependencies to solve this exact issue, but new versions won't install peerDependencies automatically when they aren't installed. To my knowledge, there's not much magic we can do in package.json to get this (very preferable) behavior. Hopefully `dedupe` should do the trick!

**Comment by BrockReece** [comment]
I'm wondering if we can do a conditional import in our plugin based on whether the user passed through their own version of moment or not. At the moment this isn't the case. Do you want me to have a look?

**Comment by brockpetrie** [comment]
@BrockReece Good idea!

**Comment by TarekAlQaddy** [comment]
@BrockReece  Yes I think that would solve it

**Comment by BrockReece** [comment]
Hi everyone, I have put in a PR for this, could do with some end to end testing though.

**Comment by brockpetrie** [comment]
I went ahead and merged since it seemed to do the trick for me locally. Can revisit if further testing shows otherwise!

**Comment by BrockReece** [comment]
I have released this change in v3.2.0
